### DISCUSSION TITLE
Sort by relation of an explicit table's field

### DISCUSSION THREAD

**Discussion by digitex** [prisma/prisma#8309] [Q&A] [answered]
Is it possible to sort `orderBy` a field of an explicit many-to-many relation?

For example, I have a `User` and a `Post` model like so, with an explicit `UserLikedPost` model.

```prisma
model User {
  id         Int             @id @default(autoincrement())
  name       String
  likedPosts UserLikedPost[]
}

model Post {
  id      Int             @id @default(autoincrement())
  title   String
  likedBy UserLikedPost[]
}

model UserLikedPost {
  createdAt DateTime @default(now())
  user      User     @relation(fields: [userId], references: [id])
  post      Post     @relation(fields: [postId], references: [id])
  userId    Int
  postId    Int
}
```

How would I write a Prisma query to get all posts liked by user with id `someId` and sorted by `createdAt`? I tried the following:

```ts
await prisma.post.findMany({
  where: {
    likedBy: {
      some: {
        user: {
          id: someId
        }
      }
    }
  },
  orderBy: {
    likedBy: {
      createdAt: 'desc'
    }
  }
})
```

 But got this error: Unknown arg `createdAt` in orderBy.likedBy.createdAt for type UserLikedPostOrderByRelationAggregateInput.

Well `createdAt` is not a field of `User` model so that makes sense but how to sort by `createdAt` of the `UserLikedPost` model?

Thanks!

**Comment by ryands17** [comment, marked as answer]
@digitex 👋 

Currently this isn't supported so it would be great if you could add a 👍 to this request so that we can look at the priority for this: https://github.com/prisma/prisma/issues/5837

The only workaround I can think of is a [raw query](https://www.prisma.io/docs/concepts/components/prisma-client/raw-database-access) for now.

**Reply by digitex** [reply]
Noted and thanks!

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
